File: sdkwork-backend-sdk-java/src/main/java/com/sdkwork/backend/api/ProjectApi.java
package com.sdkwork.backend.api;

import com.sdkwork.backend.http.HttpClient;
import com.sdkwork.backend.model.*;
import java.util.List;
import java.util.Map;

public class ProjectApi {
    private final HttpClient client;
    
    public ProjectApi(HttpClient client) {
        this.client = client;
    }

    /** Update an existing project */
    public PlusApiResultPlusProjectVO update(PlusProjectForm body) throws Exception {
        return (PlusApiResultPlusProjectVO) client.put(ApiPaths.backendPath("/project"), body);
    }

    /** Create a new project */
    public PlusApiResultPlusProjectVO create(PlusProjectForm body) throws Exception {
        return (PlusApiResultPlusProjectVO) client.post(ApiPaths.backendPath("/project"), body);
    }

    /** Get projects by page */
    public PlusApiResultPagePlusProjectVO listByPage(QueryListForm body, Map<String, Object> params) throws Exception {
        return (PlusApiResultPagePlusProjectVO) client.post(ApiPaths.backendPath("/project/list"), body, params);
    }

    /** Get all projects */
    public PlusApiResultListPlusProjectVO listAllEntities(QueryListForm body) throws Exception {
        return (PlusApiResultListPlusProjectVO) client.post(ApiPaths.backendPath("/project/list/all"), body);
    }

    /** Get a project by ID */
    public PlusApiResultPlusProjectVO getById(String id) throws Exception {
        return (PlusApiResultPlusProjectVO) client.get(ApiPaths.backendPath("/project/" + id + ""));
    }

    /** Delete a project */
    public PlusApiResultBoolean delete(String id) throws Exception {
        return (PlusApiResultBoolean) client.delete(ApiPaths.backendPath("/project/" + id + ""));
    }
}

File: sdkwork-backend-sdk-kotlin/src/main/kotlin/com/sdkwork/backend/api/ProjectApi.kt
package com.sdkwork.backend.api

import com.sdkwork.backend.*
import com.sdkwork.backend.http.HttpClient

class ProjectApi(private val client: HttpClient) {

    /** Update an existing project */
    suspend fun update(body: PlusProjectForm): PlusApiResultPlusProjectVO? {
        return client.put(ApiPaths.backendPath("/project"), body) as? PlusApiResultPlusProjectVO
    }

    /** Create a new project */
    suspend fun create(body: PlusProjectForm): PlusApiResultPlusProjectVO? {
        return client.post(ApiPaths.backendPath("/project"), body) as? PlusApiResultPlusProjectVO
    }

    /** Get projects by page */
    suspend fun listByPage(body: QueryListForm? = null, params: Map<String, Any>? = null): PlusApiResultPagePlusProjectVO? {
        return client.post(ApiPaths.backendPath("/project/list"), body, params) as? PlusApiResultPagePlusProjectVO
    }

    /** Get all projects */
    suspend fun listAllEntities(body: QueryListForm? = null): PlusApiResultListPlusProjectVO? {
        return client.post(ApiPaths.backendPath("/project/list/all"), body) as? PlusApiResultListPlusProjectVO
    }

    /** Get a project by ID */
    suspend fun getById(id: String): PlusApiResultPlusProjectVO? {
        return client.get(ApiPaths.backendPath("/project/$id")) as? PlusApiResultPlusProjectVO
    }

    /** Delete a project */
    suspend fun delete(id: String): PlusApiResultBoolean? {
        return client.delete(ApiPaths.backendPath("/project/$id")) as? PlusApiResultBoolean
    }
}
